ssl/d1_lib.c

```diff
@@ -246,16 +253,34 @@ void dtls1_start_timer(SSL *s)
     }
 #endif
 
-    /* If timer is not set, initialize duration with 1 second */
+    /*
+     * If timer is not set, initialize duration with 1 second or
+     * a user-specified value if the timer callback is installed.
+     */
     if (s->d1->next_timeout.tv_sec == 0 && s->d1->next_timeout.tv_usec == 0) {
-        s->d1->timeout_duration = 1;
+
+        if (s->d1->timer_cb != NULL)
+            s->d1->timeout_duration_ms = s->d1->timer_cb(s, 1000);
+        else
+            s->d1->timeout_duration_ms = 1000;
     }
 
     /* Set timeout to current time */
     get_current_time(&(s->d1->next_timeout));
 
     /* Add duration to current time */
-    s->d1->next_timeout.tv_sec += s->d1->timeout_duration;
+
+    duration.tv_sec  =  s->d1->timeout_duration_ms / 1000;
+    duration.tv_usec = (s->d1->timeout_duration_ms % 1000) * 1000;
```

Comment:
If switching to usec, then check to see if value is above 1000000, and to the modulo and division then, otherwise, simple assignment. (But this is me coming from my routing background.)